### PR TITLE
Add comment about the importance of AsyncSpinner

### PR DESCRIPTION
**Pull request by samarth-robo**
will prevent others from wasting hours trying to debug why the code snippets they borrowed from this tutorial do not work. Covers
- the case mentioned [in this ROS answer](https://answers.ros.org/question/302283/failed-to-fetch-current-robot-state/?answer=348497#post-id-348497)
- the common case where `move_group` and `joint_model_group` are class variables and initialized in a constructor

### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by felixvd — APPROVED** [review]
It isn't very systematic, but it doesn't hurt either.

**Comment by mlautman** [comment]
Howdy @samarth-robo long time no see :)

You need to remove trailing whitespace as per our clang format rules before we can merge this

**Comment by samarth-robo** [comment]
hello @mlautman :) I synced this branch with the upstream master, and applied `clang-format-10` as described at https://moveit.ros.org//documentation/contributing/code/. Hopefully it is good to go now, but let me know if something more needs to be done.

**Review by rhaschke — APPROVED** [review]

**Pull request merged**
